lib/include/jxl/decode.h

```diff
@@ -162,14 +162,19 @@ typedef enum {
    */
   JXL_DEC_NEED_IMAGE_OUT_BUFFER = 5,
 
-  /** Informative event by JxlDecoderProcessInput: JPEG reconstruction buffer is
-   * too small for reconstructed JPEG codestream to fit.
-   * JxlDecoderSetJPEGBuffer must be called again to make room for remaining
-   * bytes. This event may occur multiple times after
-   * JXL_DEC_JPEG_RECONSTRUCTION
+  /** The JPEG reconstruction buffer is too small for reconstructed JPEG
+   * codestream to fit. JxlDecoderSetJPEGBuffer must be called again to make
```

Comment:
The implementation of the API would ideally output at least some data before calling this function again, but it shouldn't be required to by documentation and user should be prepared to handle the case where it doesn't, to be more generic.

That doesn't mean we shouldn't make the implementation do it because it's more effective when it outputs something, than when it continuously asks the user for more bytes without outputting anything, but no promises in the header should be made.

Speaking about this, I added comment to JxlDecoderSetBoxBuffer and JxlDecoderSetJPEGBuffer that specifies if you released a buffer, the next time  you set it, you should not include the already-output bytes in the buffer.